### PR TITLE
For #7346 update lint-baseline.xml and remove activity_addons.xml as unused resource

### LINKED ISSUE #7436: Enable `UnusedResources` lint check to decrease APK size
Our `res` dir is 718.9 KB & our `resources.arsc` (the compiled resources) is 2.1 MB: of these 213 resources are marked as "Unused" by lint. Removing these unused resources may save us a non-trivial amount of APK size.

It's easy to enable UnusedResources to prevent further regressions though one challenge we may face is that Strings may be added by string imports that are not yet used.

It's also easy to remove unused resources: run "Analyze" -> "Run Inspection by name" & type in "Unused Resources" in Android Studio. When it completes, click to remove the unused resources.

### Acceptance criteria
- Enable `UnusedResources` lint check
  - N.B.: we need to consider what will happen to String imports. Should we add a blanket, "Keep unused" to them? But then how will we know when they're unused and when we can remove them? Perhaps we should add the blanket and add a follow-up to address later
- Remove all `UnusedResources` or update baseline (so there are no failures) & file a follow-up to do so
  - Consider whether all unused strings should actually be removed

### PR DESCRIPTION
**Pull request by RotBolt**
### Pull Request checklist

- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [X] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

Closes #7436 
PR includes adding warnings for `Unused Errors`  in `lint-baseline.xml`
And remove some warnings for `Unused resources`

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by RotBolt** [comment]
cc @mcomella  I have updated `lint-baselin.xml` and remove 3 errors for now for `Unused resources`.
Will make another PR for removing left out `Unused resources` as discused

File: app/src/main/res/values/arrays.xml
Comment (mcomella) [inline app/src/main/res/values/arrays.xml:37]: Sometimes when lint says a resource is unused, it's wrong because the code is referenced in a tricky way: this is one of those cases. To figure this out, I used our code search tool, https://searchfox.org/mozilla-mobile/source/, to see where this may be referenced from. I found it was [defined and read](https://searchfox.org/mozilla-mobile/search?q=mozac_url_schemes_blocklist&path=&case=false&regexp=false) in our main application library, `android-components`, that fenix consumes. The android-components repo can be found here: https://github.com/mozilla-mobile/android-components/.

I verified that it was still in use by:
- Checking out a commit without your changes to remove this code
- Adding a call to `SafeUrl.stripUnsafeUrlSchemes` in `HomeActivity.onCreate` (`stripUnsafeUrlSchemes` is the code that accesses the resources; it's called in the code normally but I had difficulty reproducing it that way)
- Starting the debugger with a breakpoint on that method ^
- Verifying the definition from fenix is used in the debugger rather than the one in android-components

The reason Android Lint was unable to detect this error is because it doesn't search an application's libraries to determine if code is in use. And since the resource files get merged, fenix's definition overrides the one from android-components.

---

As for what we should do about this issue, we should:
1. leave the code in the app
2. Explicitly ignore the `UnusedResources` error for `mozac_url_schemes_blocklist` (and the other strings if necessary but it might be smart enough not to)
3. Add a comment to explain why we explicitly ignore the error (i.e. the code that accesses this is defined in ac so the linter can't detect it).

In the future, when removing references to unused resources, it's good to double-check they're not used from `android-components` by using the searchfox tool I mentioned above. Naturally, I didn't expect you to know about this in advance so thanks for understanding!

**Review by mcomella — CHANGES_REQUESTED** [review]
Thanks for making this change! There's an issue with one of the unused resources removed but besides that this PR looks good!

File: app/src/main/res/values/arrays.xml
Comment (RotBolt) [reply app/src/main/res/values/arrays.xml:37]: Ok sure, will make the required changed

Comment (RotBolt) [reply app/src/main/res/values/arrays.xml]: This has been resolved . And thanks! I wil use the code search tool to fix the `unused resources` in subsequent PR

**Review by mcomella — APPROVED** [review]
This looks good to me – thanks! Let's land it!

One nitpick that we can address for your next PR: your commit summary is
> mozilla-mobile#7346 update lint-baseline.xml and remove activity_addons.xml as unused resource

Our preferred format is:
> For #7346: update lint-baseline.xml and remove activity_addons.xml as unused resource

But again, I think this is good for now: you've got the bug number in there and that's what matters the most. Good commit message, btw! :)

**Comment by RotBolt** [comment]
Ah, I thought `For` is only for PR message. Will add it for subsequent PRs

**Comment by mcomella** [comment]
To give you context on how we get this merged, for security reasons, we only run partial CI for PRs outside of our GitHub organization. To run the full CI, someone inside our GH org needs to open a PR with the same commits (e.g. I opened https://github.com/mozilla-mobile/fenix/pull/20592). Once the full CI runs and is green, we can land it so that's what we're waiting on now.

**Comment by jonalmeida** [comment]
I'm going restart the CI build on this PR (by closing and re-opening it) so we can see if that helps figure out why Mergify is paused here, but also to unblock the queue of PRs that are being held up.

**Comment by mcomella** [comment]
@RotBolt We recently made changes to our CI system and it seems the changes we made may have broke our workflow to merge PRs from folks outside of our GH organization (I filed https://github.com/mozilla-mobile/fenix/issues/20598 for the issue). Your commit will land either as part of this PR or with my copy https://github.com/mozilla-mobile/fenix/pull/20592 (which has your exact commit & authorship information) but it may take until Monday or Tuesday for it to land. Sorry for any confusion all the open/closing of this PR might have created! 😞

**Comment by mcomella** [comment]
@Mergifyio refresh

**Comment by mcomella** [comment]
@Mergifyio refresh

**Comment by mcomella** [comment]
@Mergifyio rebase

**Comment by RotBolt** [comment]
Thanks @mcomella for explaining the issue

**Comment by RotBolt** [comment]
For removing the unused resources, I should chat on Riot in Fenix room right ? @mcomella

**Comment by pocmo** [comment]
Force-merged the PR to unblock our queue (I hope. Not sure yet why this has happened..)

**Comment by mcomella** [comment]
@RotBolt I filed https://github.com/mozilla-mobile/fenix/issues/20634 to remove the suppressions in the lint baseline file. When making new commits to remove them, please specify that issue number (e.g. "For #20634:").

> For removing the unused resources, I should chat on Riot/matrix in Fenix room right ? @mcomella

If you want to chat with the team, then yes – that's the channel to discuss on. I'm not sure how many folks scan that channel so it's a good practice to @ mention someone with your messages (e.g. @mcomella would message me).

**Pull request merged**
